zerver/test_signup.py

```diff
@@ -40,8 +40,10 @@ def fetch(self, method, urls, expected_status):
         for url in urls:
             if method == "get":
                 response = self.client.get(url)
-            else:
+            elif method == "post":
                 response = self.client.post(url)
+            else:
+                response = self.client.put(url)
```

Comment:
Small suggestion here: add `elif method == "put"` and put under the else clause raising an exception, so if we add a new one it doesn't fail silently.

I suppose there's the also the fancy method which would achieve the same goal of replacing the whole if statement with something like `getattr(self.client, method)(url)`, though if we do that we'll probably want a comment translatiing it for clarify along the lines of`# e.g. self.client.post(url) if method is "post"`.